### PR TITLE
Usage of covariates in ST pipelines

### PR DESCRIPTION
**Pull request by c-yuwei**
This pull request will solve the conditional usage for USE_COVARIATES and CORRELATION_RULE by postprocessing the functionality.

When the USE_COVARIATES is checked, only partial Pearson and partial Spearman are available.
When the USE_COVARIATES is unchecked, only Pearson, Spearman, and Kendall are available.

See testing video:
https://user-images.githubusercontent.com/30530538/156452885-2efacf82-5b5c-41c9-8967-109df1ca7162.mp4

### CODE REVIEW

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
* Use only one field (Pearson/Spearman with convariates instead of partial).
* Remove PPs
* Check Correlation and its docs

**Comment by c-yuwei** [comment]
- Removed "USE_COVARIATES" in all related files with the following pipelines
ST, BUD, BUT, MP_ST, MP_BUD, and MP_BUT
- Tested with all abovementioned pipelines.

File: braph2genesis/src/analysis/Correlation.m [hunk 1]
```diff
@@ -8,8 +8,9 @@
     % negative rule.
     %
     % Correlation properties (Constant)
-    %  Correlation_Rule_List        - pearson, spearman, kendall, partial
-    %                                 pearson, partial spearman
+    %  Correlation_Rule_List        - Pearson, Spearman, Kendall, Pearson
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/Correlation.m:11]: capital

File: braph2genesis/src/analysis/Correlation.m [hunk 2]
```diff
@@ -18,7 +19,7 @@
     % See also Analysis, Measurement.   
     
     properties (Constant)
-        CORRELATION_RULE_LIST = {'pearson', 'spearman', 'kendall', 'partial pearson', 'partial spearman'};
+        CORRELATION_RULE_LIST = {'Pearson', 'Spearman', 'Kendall', 'Pearson with covariates', 'Spearman with covariates'};
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/Correlation.m:22]: PEARSON = 'Pearson';
SPEARMAN = 'Spearman';
... 'Kendall'
PEARSON_CV = 'Pearson with covariates'
'Spearman with covariates'
CORRELATION_RULE_LIST = {Correlation.PEARSON ...};

File: braph2genesis/src/analysis/Correlation.m [hunk 3]
```diff
@@ -61,9 +62,9 @@
                     [A, P] = corr(data, 'Type', 'Spearman');
                 case 'kendall'
```
Comment (giovannivolpe) [inline braph2genesis/src/analysis/Correlation.m:63]: case Correlation.KENDALL ...

File: braph2genesis/pipelines/structural/example_ST_WU.m [hunk 4]
```diff
@@ -29,9 +29,8 @@
 gr2 = im_gr2.get('GR');
 
 %% Analysis ST WU
-use_covariates_in_analysis = true;
-a_WU1 = AnalyzeGroup_ST_WU('GR', gr1, 'CORRELATION_RULE', 'partial pearson', 'USE_COVARIATES', use_covariates_in_analysis);
-a_WU2 = AnalyzeGroup_ST_WU('GR', gr2, 'CORRELATION_RULE', 'partial pearson', 'USE_COVARIATES', use_covariates_in_analysis);
+a_WU1 = AnalyzeGroup_ST_WU('GR', gr1, 'CORRELATION_RULE', 'Pearson with covariates');
```
Comment (giovannivolpe) [inline braph2genesis/pipelines/structural/example_ST_WU.m:32]: change

File: braph2genesis/pipelines/structural/_AnalyzeGroup_ST_WU.gen.m [hunk 5]
```diff
@@ -49,7 +44,7 @@
     atlas = gr.get('SUB_DICT').getItem(1).get('BA');
 end
 
-if a.get('USE_COVARIATES')
+if any(strcmp(a.get('CORRELATION_RULE'), {'Pearson with covariates', 'Spearman with covariates'}))
```
Comment (giovannivolpe) [inline braph2genesis/pipelines/structural/_AnalyzeGroup_ST_WU.gen.m:47]: change

**Review by giovannivolpe — CHANGES_REQUESTED** [review]

File: braph2genesis/pipelines/structural/_AnalyzeGroup_ST_WU.gen.m (hunk 5 above)
Comment (c-yuwei) [reply braph2genesis/pipelines/structural/_AnalyzeGroup_ST_WU.gen.m]: done

File: braph2genesis/pipelines/structural/example_ST_WU.m (hunk 4 above)
Comment (c-yuwei) [reply braph2genesis/pipelines/structural/example_ST_WU.m]: done

File: braph2genesis/src/analysis/Correlation.m (hunk 3 above)
Comment (c-yuwei) [reply braph2genesis/src/analysis/Correlation.m]: done

File: braph2genesis/src/analysis/Correlation.m (hunk 2 above)
Comment (c-yuwei) [reply braph2genesis/src/analysis/Correlation.m]: done

File: braph2genesis/src/analysis/Correlation.m (hunk 1 above)
Comment (c-yuwei) [reply braph2genesis/src/analysis/Correlation.m]: done

**Review by giovannivolpe — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**